firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/DataCollectionArbiter.java

```diff
@@ -107,13 +116,29 @@ public boolean isAutomaticDataCollectionEnabled() {
   }
 
   @SuppressLint({"CommitPrefEdits", "ApplySharedPref"})
-  public void setCrashlyticsDataCollectionEnabled(boolean enabled) {
-    crashlyticsDataCollectionEnabled = enabled;
-    crashlyticsDataCollectionExplicitlySet = true;
-    sharedPreferences.edit().putBoolean(FIREBASE_CRASHLYTICS_COLLECTION_ENABLED, enabled).commit();
+  public synchronized void setCrashlyticsDataCollectionEnabled(Boolean enabled) {
```

Comment:
What is this method synchronizing that was missed before?

Reply:
Yes looks like it was missed before.

Reply:
My question was more the *what* is being synchronized, rather than "was this missed". I'm not 100% sure this needs to be synchronized, but if some state in here does need to be, we should probably be more explicit about it and try to use a synchronized block with a named lock for clarity.

Reply:
Talked offline about this: going to leave as it is for now and potentially polish it in a subsequent commit.